### PR TITLE
NixOS: Document impurity issues with boot.binfmt.emulatedSystems

### PR DESCRIPTION
**Pull request by ck3d**
###### Motivation for this change
For cross compiled packages, we have to take care boot.binfmt.emulatedSystems is not set. This PR adds more some note to the NixOS option.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/system/boot/binfmt.nix [hunk 1]
```diff
@@ -248,6 +248,8 @@ in {
         description = ''
           List of systems to emulate. Will also configure Nix to
           support your new systems.
+          Be careful as a build can execute all emulated systems at the same time.
```
Comment (markuskowa) [inline nixos/modules/system/boot/binfmt.nix:251]: Maybe this is better:
```suggestion
          Warning: the builder can execute all emulated systems within the same build, which introduces impurities in the case of cross compilation.
```

File: nixos/modules/system/boot/binfmt.nix [hunk 2]
```diff
@@ -248,7 +248,7 @@ in {
         description = ''
           List of systems to emulate. Will also configure Nix to
           support your new systems.
-          Be careful as a build can execute all emulated systems at the same time.
+          Warning: the builder can execute all emulated systems within the same build, which introduces impurities in the case of cross compilation.
           In case of cross compilation this introduces impurity.
```
Comment (markuskowa) [inline nixos/modules/system/boot/binfmt.nix:252]:
```suggestion
```

File: nixos/modules/system/boot/binfmt.nix [hunk 3]
```diff
@@ -248,6 +248,7 @@ in {
         description = ''
           List of systems to emulate. Will also configure Nix to
           support your new systems.
+          Warning: the builder can execute all emulated systems within the same build, which introduces impurities in the case of cross compilation.
```
Comment (SuperSandro2000) [inline nixos/modules/system/boot/binfmt.nix:251]:
```suggestion
          Warning: the builder can execute all emulated systems within the same build, which can introduce impurities in the case of cross compilation.
```

Comment (ck3d) [reply nixos/modules/system/boot/binfmt.nix:251]: It definitely does introduce impurity.

**Pull request merged**